### ISSUE TITLE
SocketServer.py error while trying to run more than three guests concurrently 

### ISSUE THREAD

**Issue by liranfar** [cuckoosandbox/cuckoo#1362]
I'm trying to run more than one vm concurrently and I get the following error: 

Exception happened during processing of request from ('192.168.56.101', 35699)
Traceback (most recent call last):
  File "/usr/lib/python2.7/SocketServer.py", line 596, in process_request_thread
    self.finish_request(request, client_address)
  File "/usr/lib/python2.7/SocketServer.py", line 331, in finish_request
    self.RequestHandlerClass(request, client_address, self)
  File "/usr/lib/python2.7/SocketServer.py", line 652, in __init__
    self.handle()
  File "/home/farage/Desktop/Final-Project-Cuckoo-Droid/cuckoo/lib/cuckoo/core/resultserver.py", line 223, in handle
    self.protocol.close()
AttributeError: 'NoneType' object has no attribute 'close' object has no attribute 'close'﻿

There are many errors like that, I just copy one of them...

**Comment by yeejay** [comment]
Just a tip: it helps to post more than just the error.  Configs, hypervisor, etc. help troubleshoot.  Also, run cuckoo.py with the debug flag.

When you say you're trying to run concurrent vms, do you mean having more than one cuckoo instance trying to process files at the same time?  Are you receiving this error if running only one vm?

**Comment by liranfar** [comment]
Hi and thanks for the tips
Im running 1.2 version with **one** instance of cuckoo. Im recieving the error from all the guests. config file are set to headless mode , 4 VMs, virtualBox,
Host OS: ubuntu 16.04
guest OS: ubuntu 12.04
I run cuckoo in debug mode but there wasn't any relevant information besides those errors.
I get the mentioned error both in guest and host.
@yeejay is it recommended to run several cuckoo nodes concurently? I didn't try that, how can I make that work?

**Comment by yeejay** [comment]
Is there a reason you're running an old version of cuckoo?  I think your issue was resolved long ago in one of the development branches

**Comment by liranfar** [comment]
yes, I am working with cuckoo-droid extension for running apks

**Comment by liranfar** [comment]
Upgrading to 2.0 version fixed this issue. Thanks a lot!

**Comment by liranfar** [comment]
After updating to latest dev version (2.0), the maximum guests I can run concurrently without any error 3.
running 4+ leads to connection reset error "broken pipe" in the guests POSTS requests screen (socket.server.py error). seems that the host can't handle more than that. I've read that there is maximum size buffer and I guess that it related to that issue, is that changeable?
just for reminding: I'm running apk on android emulator. using androguard and droidmon. 
I've also tried to run several cuckoo host process, just running cuckoo.py in several windows after I edited the configuration files and set 2 different result_server port respectively to 3 guests each but it didnt run well. :cry:
Any suggestions?
